### DISCUSSION TITLE
Chinese ble bbq probe

### DISCUSSION THREAD

**Discussion by TurboTronix** [theengs/decoder#149] [New devices/things integration]
I have the following device that I'm trying to integrate:

![Screenshot_20220708-213523.png](https://user-images.githubusercontent.com/7071554/178087072-5bbc9bc6-ef2d-4f78-9fc4-a93f20c4981d.png)

![Screenshot_20220708-213618.png](https://user-images.githubusercontent.com/7071554/178087075-7dbae4b4-a285-49ae-993d-5aeaaeed2fee.png)

**Comment by TurboTronix** [comment]
Temperature info:

At 34 degrees Celsius

![Screenshot_20220708-213925.png](https://user-images.githubusercontent.com/7071554/178087103-825f9c54-df41-4661-a997-f5f93beb714e.png)

**Reply by TurboTronix** [reply]
At 43 degrees Celsius:

![Screenshot_20220708-214255.png](https://user-images.githubusercontent.com/7071554/178087123-43849948-e32c-43ca-aa90-91068809bc14.png)

**Reply by TurboTronix** [reply]
At 52 degrees Celsius:

![Screenshot_20220708-214138.png](https://user-images.githubusercontent.com/7071554/178087145-7bce2f1a-1745-4304-ab3c-617db708425d.png)

**Reply by TurboTronix** [reply]
At 51 degrees Celsius:

![Screenshot_20220708-214128.png](https://user-images.githubusercontent.com/7071554/178087154-8f09a155-9145-4b97-a751-f47d80a66fd7.png)

**Comment by koenvervloesem** [comment]
Decoder is only able to decode broadcasted data, while you are showing data read while connected to the device. Can you post screenshots of what the nRF Connect app is showing about the device in its main view when you're not connected to it? If it broadcasts manufacturer data or service data and you can show the corresponding temperature, we can try to decode it.

**Reply by TurboTronix** [reply]
You mean this view:
![Screenshot_20220709-073837.png](https://user-images.githubusercontent.com/7071554/178104031-969425d3-e755-4d9b-b1e5-53fa7f0dcd69.png)

**Reply by koenvervloesem** [reply]
~Yes, do the manufacturer data change when the temperature changes?~

Oh I already see that the data are just the device's MAC address in little-endian format, so we can't do anything with it unfortunately :-(

**Reply by TurboTronix** [reply]
Only this value seems to change:
![Screenshot_20220709-085331.png](https://user-images.githubusercontent.com/7071554/178106658-d939292d-7e29-4667-be4f-20dce5867b0e.png)

**Reply by koenvervloesem** [reply]
But you can only read that value while connected, so this is currently outside the scope of Decoder.

**Reply by TurboTronix** [reply]
I see

**Reply by TurboTronix** [reply]
So i can't use anything to read this? Not even connect it to esphome via esp32?

**Reply by koenvervloesem** [reply]
There are surely other means. If you want to use ESPHome, have a look at this to try to process the characteristic's data: https://koen.vervloesem.eu/blog/connecting-to-bluetooth-low-energy-devices-in-esphome/

OpenMQTTGateway can also read characteristics: https://docs.openmqttgateway.com/use/ble.html#read-write-ble-characteristics-over-mqtt-esp32-only But you further need to process them to turn it into a temperature. Maybe someone on the [OpenMQTTGateway forum](https://community.openmqttgateway.com/) can help you to add direct support.

**Reply by TurboTronix** [reply]
![image](https://user-images.githubusercontent.com/7071554/179364474-bd60dbe7-644b-4496-a93c-8fe9e5be0362.png)

I am trying to decode the hex I am getting but it's not easy...

**Reply by DigiH** [reply]
> Only this value seems to change:
> ![Screenshot_20220709-085331.png](https://user-images.githubusercontent.com/7071554/178106658-d939292d-7e29-4667-be4f-20dce5867b0e.png)

Hi @TurboTronix,

while we cannot usually try and decode issues like that for everyone when it is not Decoder related, but for connection READ only devices, I was still a bit intrigued where the temperature might be hidden here ;)

I think it is the 3rd and 4th byte of the FB02 characteristic, then little endian/reversed, then - 400 and divided by 10 - or /10 and then - 40 ;). So for the above screenshot this would be

8E 03 - reversed 03 8E is 910 - 400 = 510 /10 = 51.0

and the other examples you provided above seem to match as well

E4 02 = 02 E4 = 740 - 400 = 340 / 10 = 34.0 ºC
3E 03 = 03 3E = 830 - 400 = 430 / 10 = 43.0 ºC
98 03 = 03 98 = 920 - 400 = 520 / 10 = 52.0 ºC

and the two bytes following are probably a second probe which works the same!?!

Hope this helps

**Reply by TurboTronix** [reply]
Wow that's crazy you decoded it.

**Reply by TurboTronix** [reply]
There are no other probes, the rest might be battery level or Bluetooth level, not really sure.

**Reply by DigiH** [reply]
Ok, just wondered as they seem similar to the 2 bytes defining the temperature of the one probe ;)

**Reply by RemcoReitsma** [reply]
I found this thread two years later and I think I know the answer; There is a second sensor on the other end of the probe to measure the ambient temp!
